pkg/registry/core/pod/rest/subresources.go

```diff
@@ -113,7 +116,20 @@ func (r *AttachREST) Connect(ctx context.Context, name string, opts runtime.Obje
 	if err != nil {
 		return nil, err
 	}
-	return newThrottledUpgradeAwareProxyHandler(location, transport, false, true, responder), nil
+	handler := newThrottledUpgradeAwareProxyHandler(location, transport, false, true, responder)
+	if utilfeature.DefaultFeatureGate.Enabled(features.ClientRemoteCommandWebsockets) {
+		// Wrap the upgrade aware handler to implement stream translation
+		// for WebSocket/V5 upgrade requests.
+		streamOptions := translator.Options{
+			Stdin:  attachOpts.Stdin,
+			Stdout: attachOpts.Stdout,
+			Stderr: attachOpts.Stderr,
+			Tty:    attachOpts.TTY,
+		}
+		streamtranslator := translator.NewStreamTranslatorHandler(location, transport, proxy.NewErrorResponder(responder), streamOptions)
```

Comment:
curious, why do we need `proxy.NewErrorResponder(responder)` ?

Reply:
We use the `ErrorResponder` in the `StreamTranslator` to send errors back to the client the same way that other proxies do (like the upgrade aware proxy). The `UpgradeAwareHandler` has the following field.
```
        // Responder is passed errors that occur while setting up proxying.                                                                                                                            
        Responder ErrorResponder                                            
```
I believe this was created because the errors can not be returned with an HTTPResponse during the connection upgrade process of the proxy.

Reply:
I see, thanks